integration/integration_test.go

```diff
@@ -275,6 +279,84 @@ func (s *IntSuite) TestAudit(c *check.C) {
 	}
 }
 
+// TestInteroperability checks if Teleport and OpenSSH behave in the same way
+// when executing commands.
+func (s *IntSuite) TestInteroperability(c *check.C) {
+	tempdir, err := ioutil.TempDir("", "teleport-")
+	c.Assert(err, check.IsNil)
+	defer os.RemoveAll(tempdir)
+	tempfile := filepath.Join(tempdir, "file.txt")
+
+	var tests = []struct {
+		inCommand   string
+		inStdin     string
+		outContains string
+		outFile     bool
+	}{
+		// 0 - echo "1\n2\n" | ssh localhost "cat -"
+		// this command can be used to copy files by piping stdout to stdin over ssh.
+		{
+			"cat -",
+			"1\n2\n",
+			"1\n2\n",
+			false,
+		},
+		// 1 - ssh -tt locahost '/bin/sh -c "mkdir -p /tmp && echo a > /tmp/file.txt"'
+		// programs like ansible execute commands like this
+		{
+			fmt.Sprintf(`/bin/sh -c "mkdir -p /tmp && echo a > %v"`, tempfile),
+			"",
+			"a",
+			true,
+		},
+		// 2 - ssh localhost tty
+		// should print "not a tty"
+		{
+			"tty",
+			"",
+			"not a tty",
+			false,
+		},
+	}
+
+	for i, tt := range tests {
+		// create new teleport server
+		t := s.newTeleport(c, nil, true)
+		defer t.Stop(true)
```

Comment:
defer will be called on the function return not on the next iteration, so that's now what you expected probably - so scope it out to a separate function to guarantee better behavior.